lib/rubocop/formatter/junit_formatter.rb

```diff
@@ -62,7 +62,10 @@ def add_testcase_element_to_testsuite_element(file, target_offenses, cop)
       end
 
       def classname_attribute_value(file)
-        file.gsub(/\.rb\Z/, '').gsub("#{Dir.pwd}/", '').tr('/', '.')
+        @classname_attribute_value_cache ||= Hash.new do |hash, key|
+          hash[key] = key.gsub(/\.rb\Z/, '').gsub("#{Dir.pwd}/", '').tr('/', '.')
```

Comment:
The original code can be updated:

```suggestion
          hash[key] = key.delete_suffix('.rb').gsub("#{Dir.pwd}/", '').tr('/', '.')
```

Reply:
I'll merge this as is for the new release and we can improve it afterwards.

Reply:
This should probably be handled by `Performance/DeleteSuffix`. This is not 100% safe, because `\Z` is different than `\z`, but in 99.9% cases I think the user meant `\z`.

There is an example of this from `Performance/EndWith` https://github.com/rubocop/rubocop-performance/blob/c155ba5f7225ca7fd72d2a3bb3912a592a7d9a18/lib/rubocop/cop/performance/end_with.rb#L17-L24
The cop does not actually handle `\Z` (the example is incorrect and should use `\z`), but it suggests that `\Z` should be replaced. 

`DeleteSuffix` is already marked as unsafe, so we can extend it with this behavior. Wdyt?

Reply:
@koic ⬆️